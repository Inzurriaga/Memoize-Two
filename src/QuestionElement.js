import React, { Component } from "react";
import brace from 'brace';
import AceEditor from 'react-ace';
import 'brace/mode/javascript';
import 'brace/theme/monokai';

export default class QuestionElement extends Component{
    constructor(){
        super();
        this.state = {
            userinput: ''
        }
    }


    render(){
        return(
            <div className="questionElement-section">
                <section className="instructions-section">
                    <h2>Instructions</h2>
                    <p>{this.props.instructions}</p>
                </section>
               <AceEditor 
               mode="javascript"
               theme="monokai"
               name="blah2"
               className="editor"
               onLoad={this.onLoad}
               onChange={this.onChange}
               fontSize={14}
               showPrintMargin={true}
               showGutter={true}
               highlightActiveLine={true}
               value={''}
               setOptions={{
               enableBasicAutocompletion: false,
               enableLiveAutocompletion: false,
               enableSnippets: false,
               showLineNumbers: true,
               tabSize: 2,
               }}
               />
               <div>
                   <p></p>
               </div>
            </div>
        )
    }
}