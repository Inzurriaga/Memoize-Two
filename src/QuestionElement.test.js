import React from 'react';
import { shallow } from 'enzyme';
import QuestionElement from "./QuestionElement"

const mockUser = {"name":"gabriel","password":"gabriel","completed":[1,2,5]};
const mockUsers = [{"name":"duy","password":"duy","completed":[1,2,5]},{"name":"gabriel","password":"gabriel","completed":[2,5]},{"name":"isaac","password":"isaac","completed":[1,2]}];
const mockSelectedQuestion = { title: "number return 5", type: "number", instructions: "return the number 5", answer: 5, startup: "function number() {\n//code here \n} \n\n number()", id: 1 };

describe("QuestionElement", ()=> {
  let wrapper; 
  beforeEach(() => {
    wrapper = shallow(
      <QuestionElement question={mockSelectedQuestion}
      users={mockUsers}
      user={mockUser} />
    )
  })

  it("should match snap shot", () => {
    expect(wrapper).toMatchSnapshot()
  })

  it("should have default state", () => {
    expect(wrapper.state()).toEqual({
      userinput: "function number() {\n//code here \n} \n\n number()",
      userOutcome: "",
      rightAnswer: "",
      correct: ""
  })
  })

  it("should change the user input on change on the code editor", () => {
    expect(wrapper.state("userinput")).toEqual( "function number() {\n//code here \n} \n\n number()")
    wrapper.find(".editor").simulate("change",  "hello")
    expect(wrapper.state("userinput")).toEqual( "hello")
  })

  it("should submit when input is click", () => {
    wrapper.find(".editor").simulate("change",  "function number() {\nreturn 5 \n} \n\n number()")
    wrapper.find("input").simulate("click")
  })
})