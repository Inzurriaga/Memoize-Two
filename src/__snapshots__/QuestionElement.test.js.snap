// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionElement should match snap shot 1`] = `
<div
  className="questionElement-section"
>
  <section
    className="instructions"
  >
    <h2>
      Instructions
    </h2>
    <p>
      return the number 5
    </p>
    <div
      className="result"
      style={
        Object {
          "border": "none",
        }
      }
    >
      <h2>
        RESULTS
      </h2>
      <p
        style={
          Object {
            "color": "none",
          }
        }
      />
      <p
        className="right-answer"
      />
      <p
        style={
          Object {
            "color": "none",
          }
        }
      />
    </div>
  </section>
  <div>
    <ReactAce
      className="editor"
      cursorStart={1}
      editorProps={Object {}}
      enableBasicAutocompletion={false}
      enableLiveAutocompletion={false}
      focus={false}
      fontSize={14}
      height="500px"
      highlightActiveLine={true}
      maxLines={null}
      minLines={null}
      mode="javascript"
      name="blah2"
      onChange={[Function]}
      onLoad={null}
      onPaste={null}
      onScroll={null}
      readOnly={false}
      scrollMargin={
        Array [
          0,
          0,
          0,
          0,
        ]
      }
      setOptions={
        Object {
          "showLineNumbers": true,
          "tabSize": 2,
        }
      }
      showGutter={true}
      showPrintMargin={true}
      style={Object {}}
      tabSize={4}
      theme="monokai"
      value="function number() {
//code here 
} 

 number()"
      width="500px"
      wrapEnabled={false}
    />
    <div
      className="submit"
    >
      <input
        onClick={[Function]}
        type="button"
        value="submit"
      />
    </div>
  </div>
</div>
`;
